protocol/common/message.go

```diff
@@ -294,10 +301,33 @@ func (m *Message) parseAudio() error {
 	return nil
 }
 
+type MentionNodeVisitor struct {
+	mentions []string
+}
+
+func (v *MentionNodeVisitor) Visit(node ast.Node, entering bool) ast.WalkStatus {
+	// only on entering we fetch, otherwise we go on
+	if !entering {
+		return ast.GoToNext
+	}
+	switch n := node.(type) {
+	case *ast.Mention:
+		v.mentions = append(v.mentions, string(n.Literal))
+	}
+	return ast.GoToNext
+}
```

Comment:
Maybe add a note to state that this is an implementation of the `github.com/status-im/markdown/ast.NodeVisitor` interface. I got confused initially trying to understand the purpose of this struct and the Visit() function